libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/controller/RouteAlternativeStateController.kt

```diff
@@ -0,0 +1,82 @@
+package com.mapbox.navigation.ui.app.internal.controller
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.extensions.flowRouteAlternativeObserver
+import com.mapbox.navigation.ui.app.internal.Action
+import com.mapbox.navigation.ui.app.internal.State
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.utils.internal.ifNonNull
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class RouteAlternativeStateController(private val store: Store) : StateController() {
+    init {
+        store.register(this)
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+        this.mapboxNavigation = mapboxNavigation
+
+        coroutineScope.launch {
+            mapboxNavigation.flowRouteAlternativeObserver().collect { pair ->
+                val routeProgress = pair.first
+                val alternatives = pair.second
+                val currentRoutes = mapboxNavigation.getNavigationRoutes()
+                val primaryRoute = currentRoutes.firstOrNull()
+                ifNonNull(primaryRoute) { routePrimary ->
+                    val isPrimaryRouteOffboard = routePrimary.origin == RouterOrigin.Offboard
+                    val offBoardAlternatives = alternatives.filter {
+                        it.origin == RouterOrigin.Offboard
+                    }
+                    val updatedRoutes = mutableListOf<NavigationRoute>()
+                    when {
+                        isPrimaryRouteOffboard -> {
+                            updatedRoutes.add(routePrimary)
+                            updatedRoutes.addAll(offBoardAlternatives)
+                        }
+                        isPrimaryRouteOffboard.not() && offBoardAlternatives.isNotEmpty() -> {
+                            updatedRoutes.addAll(offBoardAlternatives)
+                        }
+                        else -> {
+                            updatedRoutes.add(routePrimary)
+                            updatedRoutes.addAll(alternatives)
+                        }
+                    }
+                    val currentLegIndex = routeProgress.currentLegProgress?.legIndex ?: 0
+                    when (store.state.value.navigation) {
+                        is NavigationState.RoutePreview -> {
```

Comment:
This check can be removed, because you don't set the routes during route preview, so you won't receive any alternatives.